scripts/metamath-extract.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import argparse
+import itertools
+
+from ml.metamath.ast import (  # noqa: TC002
+    Application,
+    AxiomaticStatement,
+    ConstantStatement,
+    DisjointStatement,
+    FloatingStatement,
+    Metavariable,
+    ProvableStatement,
+    Statement,
+    StructuredStatement,
+    Terms,
+    VariableStatement,
+)
+from ml.metamath.backend import NullBackend, StandaloneFileBackend
+from ml.metamath.composer import Composer, Theorem  # noqa: TC002
+from ml.metamath.parser import load_database
+
+
+def get_metamath_symbols(terms: Terms) -> set[str]:
+    ret: set[str] = set()
+    for term in terms:
+        if isinstance(term, Application):
+            ret = ret.union({term.symbol}, get_metamath_symbols(term.subterms))
+    return ret
+
+
+def compressed_proof_get_lemmas(proof: str) -> tuple[str, ...]:
+    lemmas_begin = proof.find('(') + 1
+    lemmas_end = proof.find(')', lemmas_begin)
+    assert 0 <= lemmas_begin < lemmas_end, 'Can only parse compressed proofs.'
+    return tuple(proof[lemmas_begin:lemmas_end].split())
+
+
+def theorem_get_disjoints(theorem: Theorem) -> tuple[DisjointStatement, ...]:
+    ret = []
+    for disjoint in theorem.context.disjoints:
+        new_vars = tuple(
+            Metavariable(var) for var in theorem.statement.get_metavariables() if var in disjoint.get_metavariables()
+        )
+        if new_vars:
+            ret += [DisjointStatement(new_vars)]
+    return tuple(ret)
+
+
+def copy_with_context(
```

Comment:
Maybe a different name to not confuse with ML contexts?

Reply:
Ok, I see that `.context` is also used in the api, i take it back.

Reply:
I'd like to rename context to frame in the long run, since that's what Metamath manual calls it.